### DISCUSSION TITLE
Using Adsense in Next.js 11

### DISCUSSION THREAD

**Discussion by sandeepbojan** [vercel/next.js#27675] [Help]
### What example does this report relate to?

google adsense

### What version of Next.js are you using?

11

### What version of Node.js are you using?

14

### What browser are you using?

chrome

### What operating system are you using?

Mac OS

### How are you deploying your application?

next start

### Describe the Bug

Hey,

I'm currently trying to add adsense ads to my website. So here is what I've already done.

    <script data-ad-client="ca-pub-****"
            type='text/javascript'
            async
            src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js" />

import React from "react";
export default class GoogleAdSense extends React.Component {

  componentDidMount () {
    if(typeof window !== 'undefined'){
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    }
  }

  render () {
    let adTest = process.env.ORION_ENV != 'production' ? {"data-adtest":"on"} : {}

    return (
      <ins className='adsbygoogle'
        style={{ display: 'block'}}
        data-ad-client="ca-pub-****"
        data-ad-slot="slot"
        data-ad-format='auto'
        { ... adTest }
        data-full-width-responsive="true"
      />
    );
  }
}

It used to work perfectly fine with NextJS 9. But, now after the upgrade, I am getting the below error:

TypeError: Cannot read property 'type' of null
fnext  |     at /app/.next/server/chunks/6859.js:789:13
fnext  |     at /app/node_modules/react/cjs/react.development.js:1067:17
fnext  |     at mapIntoArray (/app/node_modules/react/cjs/react.development.js:964:23)
fnext  |     at mapIntoArray (/app/node_modules/react/cjs/react.development.js:1004:23)
fnext  |     at Object.mapChildren [as map] (/app/node_modules/react/cjs/react.development.js:1066:3)
fnext  |     at Head.makeStylesheetInert (/app/.next/server/chunks/6859.js:788:36)
fnext  |     at /app/.next/server/chunks/6859.js:798:36
fnext  |     at /app/node_modules/react/cjs/react.development.js:1067:17
fnext  |     at mapIntoArray (/app/node_modules/react/cjs/react.development.js:964:23)
fnext  |     at mapIntoArray (/app/node_modules/react/cjs/react.development.js:1004:23)

### Expected Behavior

I have been using this script for rendering ads in my website

### To Reproduce

import React from 'react'
import Document, { Html, Head, Main, NextScript } from 'next/document'

export default class MyDocument extends Document {
  render () {
    return (
      <Html>
        <Head>
          <script data-ad-client="ca-pub-****"
            type='text/javascript'
            async
            src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

**Comment by leodr** [comment]
You can try using the new [`<Script />` component](https://nextjs.org/docs/basic-features/script) provided by Next.js.

**Reply by sandeepbojan** [reply]
Thanks for the suggestion. I tried importing the Script from next/script and using as below. Still seeing the same above error:

<Script data-ad-client="ca-pub-****" type='text/javascript' async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js" />

Could you please let me know If I am missing anything?

**Reply by leodr** [reply]
Looks like everything is correct. I can't help you with this then, sorry

**Reply by sandeepbojan** [reply]
Thank you
